File: app/src/main/java/org/stepic/droid/core/presenters/contracts/CoursesView.java
package org.stepic.droid.core.presenters.contracts;

import org.jetbrains.annotations.NotNull;
import org.stepic.droid.model.Course;

import java.util.List;

public interface CoursesView {
    void showLoading();

    void showEmptyCourses();

    void showConnectionProblem();

    void showCourses (@NotNull List<Course> courses);
}

File: app/src/main/java/org/stepic/droid/core/presenters/contracts/CoursesView.kt
package org.stepic.droid.core.presenters.contracts

import org.solovyev.android.checkout.Sku
import org.stepik.android.domain.course_payments.model.CoursePayment
import org.stepik.android.model.Course

interface CoursesView {
    fun showLoading()

    fun showEmptyCourses()

    fun showConnectionProblem()

    fun showCourses(courses: List<Course>, skus: Map<String, Sku>, coursePayments: Map<Long, CoursePayment>)
}
